core/element_array_view.cpp

```diff
@@ -48,17 +44,52 @@ ElementArrayViewParams::ElementArrayViewParams(const scipp::index offset,
 ElementArrayViewParams::ElementArrayViewParams(
     const ElementArrayViewParams &other, const Dimensions &iterDims)
     : m_offset(other.m_offset), m_iterDims(iterDims),
-      m_dataDims(other.m_dataDims), m_bucketParams(other.m_bucketParams) {
+      m_bucketParams(other.m_bucketParams) {
   expectCanBroadcastFromTo(other.m_iterDims, m_iterDims);
-  // See implementation of ViewIndex regarding this relabeling.
-  for (const auto &label : m_dataDims.labels())
-    if (label != Dim::Invalid && !other.m_iterDims.contains(label))
-      m_dataDims.relabel(m_dataDims.index(label), Dim::Invalid);
+
+  for (scipp::index dim = 0; dim < iterDims.ndim(); ++dim) {
+    auto label = iterDims.label(dim);
+    if (other.m_iterDims.contains(label)) {
+      m_strides[dim] = other.m_strides[other.m_iterDims.index(label)];
+    } else {
+      m_strides[dim] = 0;
+    }
+  }
 }
 
 void ElementArrayViewParams::requireContiguous() const {
-  if (!m_iterDims.isContiguousIn(m_dataDims) || m_bucketParams)
+  bool is_contiguous = true;
+  if (m_bucketParams) {
+    is_contiguous = false;
+  } else {
+    scipp::index expected_stride = 1;
+    for (scipp::index dim = m_iterDims.ndim() - 1; dim >= 0; --dim) {
+      if (m_strides[dim] != expected_stride) {
+        is_contiguous = false;
+        break;
+      }
+      expected_stride *= m_iterDims[m_iterDims.label(dim)];
+    }
+  }
+  if (!is_contiguous)
     throw std::runtime_error("Data is not contiguous");
```

Comment:
`DimensionError` may be a better choice?

Reply:
Not sure. `DimensionError` is usually about extents of data and not its underlying layout.